e2e/setup-kube.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -x
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd ${REPO_ROOT}
+cd ..
+WORKSPACE=$(pwd)
+
+# Clone kubernetes into ${WORKSPACE}/kubernetes
+cd "${WORKSPACE}"
+KUBE_VERSION=${KUBE_VERSION:-master}
+git clone --depth 1 --branch ${KUBE_VERSION} https://github.com/kubernetes/kubernetes kubernetes
```

Comment:
Maybe we should assume this has already been done, either by prow with `extra_refs`, or by a user that is developing and setting up a local development workspace.

In the local-development scenario, I imagine we want the `go work...` commands (and workarounds for go.mod), but maybe these don't belong under `e2e`?

Reply:
> Maybe we should assume this has already been done, either by prow with extra_refs, or by a user that is developing and setting up a local development workspace.

yes, I think that is the best approach